payments-core/src/main/java/com/stripe/android/paymentsheet/BaseAddPaymentMethodFragment.kt

```diff
@@ -88,7 +89,13 @@ internal abstract class BaseAddPaymentMethodFragment(
                 formFragment.paramMapLiveData.observe(viewLifecycleOwner) { paramMap ->
                     sheetViewModel.updateSelection(
                         paramMap?.run {
```

Comment:
It looks like all we want here is to map a String to the actual `PaymentMethodCreateParams.Type` here, can we 
add `internal fun fromCode(code: String) = values().firstOrNull { it.code == code }` in `PaymentMethodCreateParams.Type` and do the following instead? This way we don't need create new `GenericPaymentMethod` each time(which also misses the 2nd `hasMandate` info) just to pass the String. 

```suggestion
paramMap?.run {
    PaymentMethodCreateParams.Type.fromCode(paramMap["type"] as String)
        ?.let { type ->
            PaymentMethodCreateParams(
                type,
                overrideParamMap = paramMap,
                productUsage = setOf("PaymentSheet")
            )
        }
}?.let {
    PaymentSelection.New.GenericPaymentMethod(
        selectedPaymentMethod.displayNameResource,
        selectedPaymentMethod.iconResource,
        it,
        false
    )
}
```


Also `Type.GenericPaymentMethod` collides with `PaymentSelection.New.GenericPaymentMethod`, if we decide to go this way, probably just merge `Type` and `Type.GenericPaymentMethod`

Reply:
Done.